komodo/show_version.py

```diff
@@ -0,0 +1,154 @@
+    return dict(config["MAIN"])
+
+
+def guess_komodoenv_path(release: str) -> Path:
+    """
+    Use the release name to find the 'real' release path, but from a
+    komodoenv environment. These environments overwrite the environment
+    variable $KOMODO_RELEASE and $PATH, but they store the original path
+    in the komodoenv at 'root/pyvenv.cfg'.
+
+    Args:
+        release: The name of the release, e.g. from $KOMODO_RELEASE.
+
+    Returns:
+        The path to the release, where the 'root/bin' folder is.
+    """
+    config = read_config(Path(release) / "root" / "pyvenv.cfg")
+    path, *_ = config["home"].split("/root/bin")
+    return Path(path)
+
+
+def guess_komodo_path(release: str) -> Path:
+    """
+    Use the release name to find the 'real' release path in an ordinary
+    komodo environment. E.g., the release may be something like
+    '2023.01.02-py38' but the 'real' release (where the release manifest
+    is stored) might be platform-specific, e.g. '2023.01.02-py38-rhel7'.
+    The real path is in the PATH, so we try to get it from there.
+
+    Args:
+        release: The name of the release, e.g. from $KOMODO_RELEASE.
+
+    Returns:
+        The path to the release, where the 'root/bin' folder is.
+    """
+    paths = os.environ["PATH"]
+    pattern = re.compile(rf"(.*?{release}.*?)/root/bin")
+    result = pattern.search(paths)
+    if result is not None:
+        (path,) = result.groups()
+    else:
+        message = f"Could not retrieve the path to the release {release}."
+        raise RuntimeError(message)
+    return Path(path)
+
+
+def get_version(pkg: str, manifest: Optional[Dict] = None) -> str:
+    """
+    Get the release number (or git commit hash) for a package in a
+    komodo release file. If no file is specified, the current release
+    is used. If no environment is active, the path to the release
+    must be specified.
+
+    Args:
+        pkg: The name of the package to get the version for.
+        manifest: Optional. A 'manifest' mapping packages to dicts
+            of version and maintainer.
+
+    Returns:
+        The version number or git hash of the version.
+    """
+    path = None
+    if manifest is None:
+        release = guess_release()
+        if (Path(release) / "komodoenv.conf").is_file():
+            path = guess_komodoenv_path(release)
+        else:
+            path = guess_komodo_path(release)
+        release_file = path.parts[-1]
+
+        with open(path / release_file, "rt", encoding="utf-8") as stream:
+            manifest = yaml.safe_load(stream)
+
+    package = manifest.get(pkg)
+
+    if package is None:
+        raise KeyError(f"The package {pkg} is not found in the release.")
+
+    return package.get("version")
+
+
+def parse_args(args: List[str]) -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description="Return the version of a specified package in a given "
+        "release or release 'manifest' file."
```

Comment:
Why does `manifast` need to be in ''?

Reply:
I understand. In this case, what do you think about adding this nomenclature in the [release manager manual](https://github.com/equinor/scout/blob/2e2086ec59fae3492f098b8114f1e83d9f79fcf2/release-process/README.md)? If you think it is a good idea, just creating the issue would be suffice since solving it is not exactly on the scope of this PR.

Reply:
done